base_layer/wallet/src/transaction_service/handle.rs

```diff
@@ -53,9 +53,10 @@ pub enum TransactionServiceRequest {
     GetAnyTransaction(TxId),
     SetBaseNodePublicKey(CommsPublicKey),
     SendTransaction((CommsPublicKey, MicroTari, MicroTari, String)),
+    SendOneSidedTransaction((CommsPublicKey, MicroTari, MicroTari, String)),
     CancelTransaction(TxId),
     ImportUtxo(MicroTari, CommsPublicKey, String),
-    SubmitTransaction((TxId, Transaction, MicroTari, MicroTari, String)),
+    SubmitTransaction(CompletedTransaction),
```

Comment:
Is this not leaking the CompletedTransaction (which is in the storage mod) to the public? I think it is exposing it here more than it should be

Reply:
Or perhaps CompletedTransaction should be moved out of the storage mod. That would be outside the scope of this pr

Reply:
I did not think so no, as this interface is only used for coin spilt. I will change it.

Reply:
We use the CompletedTransaction as a public model all over (sent in events, returned from accessors etc) so I suppose it should come out of storage if you think that  is not the place for it. Though while I agree the internals of storage is private it does feel like the right place to define data structures?

Reply:
Yeah, potentially a subjective choice. I would say that Models should be outside of storage since they can be used without persistence. But sounds like that is a discussion outside of the scope of this PR